packages/apollo-language-server/src/config/__tests__/config.ts

```diff
@@ -24,52 +45,66 @@ describe("ApolloConfig", () => {
   });
 
   describe("projects", () => {
-    it("creates a ClientConfig when client is present", () => {
-      const rawConfig: ApolloConfigFormat = {
+    it("uses client details when client key is present", () => {
+      const rawConfig: ApolloConfigFormat = createConfig({
         client: { service: "my-service" }
-      };
+      });
       const config = new ApolloConfig(rawConfig);
       const projects = config.projects;
-      expect(projects).toHaveLength(1);
+      expect(projects).toHaveLength(2);
       expect(projects[0].isClient).toBeTruthy();
+      expect(config.graphId).toEqual("my-service");
     });
-    it("creates a ServiceConfig when service is present", () => {
-      const rawConfig: ApolloConfigFormat = { service: "my-service" };
+    it("uses service details when service key is present", () => {
+      const rawConfig: ApolloConfigFormat = createConfig({
+        service: { name: "my-service" }
+      });
       const config = new ApolloConfig(rawConfig);
       const projects = config.projects;
 
-      expect(projects).toHaveLength(1);
+      expect(projects).toHaveLength(2);
       expect(projects[0].isService).toBeTruthy();
+      expect(config.graphId).toEqual("my-service");
     });
     it("creates multiple configs when both client and service are present", () => {
-      const rawConfig: ApolloConfigFormat = {
+      const rawConfig: ApolloConfigFormat = createConfig({
         client: { service: "my-service" },
-        service: "my-service"
-      };
+        service: { name: "my-service" }
+      });
       const config = new ApolloConfig(rawConfig);
       const projects = config.projects;
 
       expect(projects).toHaveLength(2);
       expect(projects.find(c => c.isClient)).toBeTruthy();
       expect(projects.find(c => c.isService)).toBeTruthy();
     });
+    it("creates multiple configs when nothing is present", () => {
+      const config = new ApolloConfig(createConfig({}));
```

Comment:
This definitely feels like cheating since it's using a test-only function (`createConfig`) to set up conditions for success.